### ISSUE TITLE
Simple site :: Password-protected page :: Comments cannot be posted

### ISSUE THREAD

**Issue by mdtanjid0** [Automattic/wp-calypso#72981]
### Quick summary

The comment isn't working on the Simple site with Password Protected pages.

So far the issue is replicateable with the [Shoreditch](<https://wordpress.com/theme/shoreditch/>) theme. Updated by @worldomonation: theme does not appear to matter. Reproduced on 2023 theme.

This error appears after submitting the form:

![Screen Shot on 2023-02-05 at 14:33:01](https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/cccc394c-720e-4516-b041-72d12680072d/75330627-6643-4189-92d8-84da15e166e1?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi9jY2NjMzk0Yy03MjBlLTQ1MTYtYjA0MS03MmQxMjY4MDA3MmQvNzUzMzA2MjctNjY0My00MTg5LTkyZDgtODRkYTE1ZTE2NmUxIiwiaWF0IjoxNzcxODgzMDY5LCJleHAiOjE4MDM0NTM2Mjl9.ElE8uanJqNpN31cnL1atCpT8wLMgPSWKfd82-dOjoKU)

### Steps to reproduce

1. Create a page or post.
2. If using page, add a Comment block and enable comments if required.
3. Make it password protected.
4. Submit a test comment on the page as a logged out user (eg. using Browser private mode).

### What you expected to happen

User is able to submit comments.<br>User sees a comment `Your comment is awaiting moderation. This is a preview; your comment will be visible after it has been approved.`, similar to when a comment is left on a public page/post.

### What actually happened

An error message, `Sorry, this comment could not be posted.` is shown:

![Screen Shot on 2023-02-05 at 14:33:01](https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/cccc394c-720e-4516-b041-72d12680072d/75330627-6643-4189-92d8-84da15e166e1?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi9jY2NjMzk0Yy03MjBlLTQ1MTYtYjA0MS03MmQxMjY4MDA3MmQvNzUzMzA2MjctNjY0My00MTg5LTkyZDgtODRkYTE1ZTE2NmUxIiwiaWF0IjoxNzcxODgzMDY5LCJleHAiOjE4MDM0NTM2Mjl9.ElE8uanJqNpN31cnL1atCpT8wLMgPSWKfd82-dOjoKU)

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Simple

### Logs or notes

I could replicate the issue with [Shoreditch](<https://wordpress.com/theme/shoreditch/>) theme.

I noticed that while working on this ticket 5886857-zd

**Comment by mdtanjid0** [comment]
- [ ] 5886857-zd ( needs to follow-up ).

**Comment by dpasque** [comment]
Adding some triage scrubbing notes:

.Org self-hosted site: Does not reproduce -- this is unique to WPCOM.
Simple sites: Reproduced as written
Atomic sites: Behavior is a little bit different, but is still broken. When you submit the comment, the comment area just becomes blank. The comment doesn't actually go through on the backend. 

![Screen Shot 2023-02-22 at 11 04 19 AM](https://user-images.githubusercontent.com/26772342/220703866-cac755b2-04a1-4c6e-b88f-fac05c3255d0.png)

The issue is not theme-specific, I was able to reproduce on other themes!

This definitely warrants a high priority.

**Comment by cuemarie** [comment]
Regarding the Atomic behavior, that may be already reported here: 

- https://github.com/Automattic/jetpack/issues/21012

**Comment by cuemarie** [comment]
📌 SCRUBBING : RESULT - *Replicated / Could Not Replicate / Uncertain*
- Tested on Simple – **Could Not Replicate**

📌 FINDINGS/SCREENSHOTS/VIDEO
- Tested on a Premium Plan simple site using FSE and classic themes using **Firefox**, and could not replicate:
     - Twenty Twenty Three: Comment successful
     - Twenty Twenty: Comment successful
     - Shoreditch: Comment successful

_Twenty Twenty Three_

https://github.com/Automattic/wp-calypso/assets/27249804/2bc92c9d-b371-46bc-82e6-5dbfc156b138

_Shoreditch_

https://github.com/Automattic/wp-calypso/assets/27249804/58b38aab-6b44-492c-969a-7e81e015552a


https://github.com/Automattic/wp-calypso/assets/27249804/9cf8cbbd-6e08-4a7e-80eb-760e6e2db906



📌 ACTIONS
- Requested author feedback

📌 Message to Author
- @mdtanjid0 or @dpasque - can y'all see anything I'm missing in my tests above?

**Comment by mdtanjid0** [comment]
@cuemarie Yes, you would need to check with a comment block, not the default comment. 

And then try with a private window. 

I am still able to replicate it with [Shoreditch](https://wordpress.com/theme/shoreditch/) theme.

Here is a brief video to demonstrate the issue: 

https://github.com/Automattic/wp-calypso/assets/54123436/abdfb144-caa0-4a43-b065-5a4630337d61

I hope that helps you!

**Comment by worldomonation** [comment]
📌 REPRODUCTION RESULTS
Tested on Simple – reproduced
Tested on Atomic – not tested
Replicable outside of Dotcom – not tested

I've updated the initial reproduction steps as the steps were not entirely clear to me.

📌 FINDINGS/SCREENSHOTS/VIDEO
**Simple**
Three screenshots showing the different combinations of commenting:
- top left: as unrelated user.
- top right: as the posting user.
- bottom: as a logged out user.

<img width="1680" alt="image" src="https://github.com/Automattic/wp-calypso/assets/6549265/c2709850-65c9-44fa-a9e8-903cfdaf9a7c">

📌 ACTIONS
- remove Needs triage label.
- move to Non-urgent column.
- maintain Normal priority.

📌 JUSTIFICATION

The flow to trigger this issue is very specific, resting on the fact that someone that does not have a WordPress.com account wants to leave a comment on a private post.

**Comment by worldomonation** [comment]
Addendum: when posting the comment, a HTTP 500 is returned. Nothing seems too out of the ordinary to me though.

Headers
```
Request URL:
https://<redacted>.wordpress.com/wp-comments-post.php
Request Method:
POST
Status Code:
500
Remote Address:
127.0.0.1:8080
Referrer Policy:
strict-origin-when-cross-origin
```

Payload
```
hc_post_as: guest
comment: Test 
email: <redacted>
author: Logged out user
url: 
wp_avatar: 
wp_user_id: 
wp_access_token: 
fb_avatar: 
fb_user_id: 
fb_access_token: 
submit: Post Comment
comment_post_ID: 58
comment_parent: 0
akismet_comment_nonce: f0bf9f5110
genseq: 1686547714
ak_hp_textarea: 
ak_js: 1686547714802
ak_bib: 1686547717782
ak_bfs: 1686547722728
ak_bkpc: 4
ak_bkp: 35,74;119,32;103,29;102,74;
ak_bmc: 77;76,1207;88,465;100,3421;
ak_bmcc: 4
ak_bmk: -1
ak_bck: 
ak_bmmc: 4
ak_btmc: 0
ak_bsc: 2
ak_bte: 
ak_btec: 0
ak_bmm: 949,94;667,469;1792,279;59,1294;
```
